com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/IAdditionalExport.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.resource;
+
+import java.util.Set;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.naming.QualifiedName;
+import org.eclipse.xtext.resource.IEObjectDescription;
+import org.eclipse.xtext.util.IAcceptor;
+
+
+/**
+ * Extensions for export.
+ */
+public interface IAdditionalExport {
+  /**
+   * Abstract method which computes the qualified name for a given object.
+   *
+   * @param object
+   *          context object, must not be {@code null}
+   * @return qualified name, may return {@code null}
+   */
+  QualifiedName qualifiedName(EObject object);
+
+  /**
+   * Returns the computed fingerprint for the given object.
+   *
+   * @param object
+   *          the object, must not be {@code null}
+   * @return the fingerprint, may return {@code null}
+   */
+  String computeFingerprint(EObject object);
+
+  /**
+   * Extension for fragment computation.
+   * <p>
+   * Implementations must return {@code null} if this extension is not meant to handle the EClass of the given object.
+   * </p>
+   *
+   * @param object
+   *          the object, must not be {@code null}
+   * @return the fragment segment for the object or {@code null} if this extension is not handling these objects
+   */
+  CharSequence getFragmentSegment(final EObject object);
+
+  /**
+   * Returns additional exported classes by this extension.
+   *
+   * @param resource
+   *          the resource, must not be {@code null}
+   * @return the set of {@link EClass}es, may be empty, never {@code null}
+   */
+  Set<EClass> getExportedEClasses(final Resource resource);
+
+  /**
+   * Retrieve qualified name and add it to the acceptor.
+   * <p>
+   * If at lease one of the extensions returns {@code true}, the content of the object will be processed. So return {@code true} if and only if this extension
+   * will export contained objects.
+   * </p>
+   *
+   * @param object
+   *          object
+   * @param acceptor
+   *          Acceptor to store retrieved qualified names
+   * @return true if this object's content should be processed
+   */
+  boolean doCreateEObjectDescriptions(final EObject object, final IAcceptor<IEObjectDescription> acceptor);

```

Comment:
The "do" prefix here seems a bit out of place...

Reply:
Right, from the interface point of view doesn't look right. At first I thought that keeping the name in the interface the same as the implementation called would help. But the implementation is going to be generated anyways.